Change agent user context from "ddagentuser" to "LOCAL_SERVICE"

derekwbrown opened the pull request:
### What does this PR do?

Use the built in account "LOCAL_SERVICE" instead of self-made account "ddagentuser".  

### Motivation

Self made account creation has proven less reliable that we'd like

### Additional Notes

Anything else we should know when reviewing?

derekwbrown committed:
WIP
Works on non-domain-joined machines, not on DC

Change install for `LOCAL_SYSTEM` installation

Fix registry permission setting

Add separate function addusertogroup

Change group membership
Add script for event permissions

Fix rollback scenarios

Fix rollback scenarios
Fix tests to reflect LOCAL_SERVICE setup rather than ddagentuser

Add secrets support
- change secrets helper script to set correct perms
- secrets checks for correct rights

Go fmt

Attempt to fix unit tests

Fix appveyor

derekwbrown committed:
WIP
Works on non-domain-joined machines, not on DC

Change install for `LOCAL_SYSTEM` installation

Fix registry permission setting

Add separate function addusertogroup

Change group membership
Add script for event permissions

hush-hush committed:
Make metadata collection configurable

Default metadata collection can be configured within predefined range.
'host' default collection interval has been lowered to 30min by default.

derekwbrown committed:
Fix rollback scenarios
Fix tests to reflect LOCAL_SERVICE setup rather than ddagentuser

derekwbrown committed:
Fix rollback scenarios

hush-hush committed:
Add section on how to upgrade Golang in CircleCI

derekwbrown committed:
Add secrets support
- change secrets helper script to set correct perms
- secrets checks for correct rights

Go fmt

hush-hush committed:
Update documenation following Python3 support in the agent (#4027)

Update documenation following Python3 support in the agent

hush-hush committed:
Escape message when using JSON format for the logs (#4004)

KSerrania committed:
[omnibus] Use the new RPM GPG key to sign RPM packages (#3962)

Changes the signing key we use to a new one, and updates the kitchen step-by-step instructions to match the change.

ruthnaebeck reviewed (commented):
No review needed from docs

derekwbrown committed:
Attempt to fix unit tests

derekwbrown committed:
Remove dependency on WMI. (#3992)

We've removed most of our WMI based code; the WMI service has been
linked to the agent stopping for no apparent reason

derekwbrown committed:
Fix appveyor

gbbr committed:
pkg/trace/{api,agent}: merge all container tags into _dd.containers

This change puts all container tags concatenated using the comma
separator into the a single tag called _dd.containers.

KSerrania committed:
[cmd/agent] Clean up dist/checks folder (#4032)

Removes check files in cmd/agent/dist/checks, they're being moved to the datadog_checks_base directory of integrations_core.

albertvaka committed:
[NTP] Lower log level for query errors, warn if recurrent (#4037)

Failing to reach an NTP host is not an issue, unless it's permanently unreachable.

This lowers the debug level from info to debug for sporadic errors, but logs a
warning if they happen repeatedly.

mfpierre committed:
Add back trace logs for dsd packets (#4014)

* Add back trace logs for dsd packets

leeavital committed:
Enable process agent when system-probe is enabled (#3958)

If the system probe is enabled, the process agent should enable itself since it's required to send network data to datadog.

leeavital committed:
Add networkID to process-agent checks (#4038)

p-lambert committed:
[ebpf] Refactor offset guessing code (#3969)

* [ebpf] Refactor offset guessing code

* [ebpf] Add constructor method for `eventGenerator`

* [ebpf] Add TCP handler method

* [ebpf] Nits based on review

* [ebpf] Add log entry for offset guessing

* [ebpf] Fix GETSOCKOPT system call

* [ebpf] Simplify state machine code

* [ebpf] Remove TCP linger configuration (no longer necessary)

* [ebpf] Simplify guess_offsets function

KSerrania committed:
[rtloader] Fix encoding of yaml.dump in Python 3 (#4009)

In Python 3, within rtloader, yaml.dump has 'utf-8' as its default encoding, which results in an encoded string being returned (b'<yaml>'). Since as_string doesn't support encoded strings as input, generating the list of integrations fails.

as_string has been modified to support encoded strings as input.

olivielpeau committed:
[omnibus] Upgrade packaged python 3 on Linux/macOS to 3.7.4 (#4045)

Brings the version in line with packaged python 3 version on
Windows.

albertvaka committed:
Fix configcheck crash when agent not ready (again) (#4046)

We were already checking for `ac.store == nil`, but actually the `store`
field gets initialized as soon as the AutoConfig is created, so we should
really check for `ac` itself being null.

The original fix was added in #3602

truthbk committed:
[omnibus][python2] shared build (#3554)

xlucas committed:
Run e2e tests with both python images in gitlab (#4036)

ofek committed:
Set explicit patch target (#4044)

ogaca-dd committed:
Remove home folder windows (#3994)

* Set the user profile folder during install to be able to remove it during the uninstall.

* Fix unrelated typo: First argument should be the same as in the first call of LookupAccountName.

* Add Userenv.lib.

* Delete user profile folder when removing the user.

* Remove locked user profile folder.

* Add release note.

KSerrania committed:
[CircleCI] Add doxygen doc generation step (#4011)

Add a CircleCI job which generates the rtloader documentation to check that there are no errors.

zippolyte committed:
Set errors in more places for easier troubleshooting (#4050)

p-lambert committed:
[ebpf] Add support for RTT (round-trip time) (#3989)

* [ebpf] Enable offset guessing for RTT fields

* [ebpf] Wait until socket params are stable

* [ebpf] Fix sneaky 2-bit imprecision bug

* [ebpf] Remove log message

* [ebpf] Simplify RTT correction logic

* [ebpf] Refactor connection tuple code

* [ebpf] Read tuple before handling message

* [ebpf] Add RTT fields to connection stats (eBPF program)

* [ebpf] Remove PID function arguments

* [ebpf] Keep status as an argument to `read_conn_tuple`

* [ebpf] Store RTT information in `ConnectionStats`

* [ebpf] Improve `nettop` output format

* [ebpf] Add test for RTT collection

DylanLovesCoffee committed:
blacklist linux-aws versions prior to 4.4.0-1060 (#4005)

* blacklist linux-aws versions prior to 4.4.0-1060

* refactor tests

* aws && ubuntu takes precedent and account for case insensitivity

gbbr committed:
pkg/trace/api: improve and instrument TCP connection rate limiter

This change simplifies the TCP connection rate limiter by removing the
unnecessary timeout and introducing metrics for incoming TCP connections
with a `status` tag having the potential values: `accepted`, `rejected`,
`timedout` and `errored`.

remeh committed:
Remove yum warnings while removing the Agent package. (#3781)

Since 6.12.0, the prerm and preinst scripts take care of removing the embedded integrations,
however, most of those files are installed by yum with the package.
A package manager tries to remove all the files that it has installed while installing a pkg, apt
doesn't seem to complain when some of these files are missing, but yum/rpm does, and, as
we're removing files on our own in scripts, it results with yum throwing a lot of warnings.

In order to avoid this, I've added a step in the finalize step of the package creation: it creates a
file listing every files in /opt/datadog-agent/embedded/lib/python2.7/site-packages. This file
is later used to compare what has been installed by the package and what is actually on the
filesystem, to remove only what was not installed by the package.

KSerrania committed:
Make Python 3 linter timeout configurable (#4049)

* Make Python 3 linter timeout configurable

The 4s timeout for the Python 3 linter is not enough for some
custom checks which take a really long time to be linted.
The default has been set to 8s, and the option is now configurable
in the datadog.yaml configuration file.

* int and time.Duration are incompatible types

* Address review feedback

* Log linter warnings

truthbk committed:
[tracemalloc] set runners + expose tracemalloc setting in `datadog-agent` builtin (#4043)

* [config] get correct number of workers according to setup

* [rtloader] expose builtin to get tracemalloc config

* [reno] adding release note

* [config] documenting tracemalloc_debug option

* [tracemalloc] review feedback (still missing test)

* [tracemalloc] address all config overrides within the config pkg

* [tracemalloc] adding missing builtin test

* [tracemalloc] fixing lint issues

* [tracemalloc] rename to tracemalloc_enabled

* [tracemalloc] fixing tests

truthbk committed:
[python] reintroduce pympler (#3696)

* [py][six] adding pympler interpreter memory usage stats

[six][two] fixing typo

* [dep] updating gopkg.lock

* [agent][api] fix python memory status call

[agent][api] add python build flag

[agent][api] add getPythonStatus stub when python build is not enabled

[python] fix pympler reporting - use structs for JSON structure

[agent][cmd] make it easier to query component status - python stats

* [release] build off Maxime guppy branch

* [pympler] disable on windows for now

* [rtloader] fix six naming vestiges

[rtloader][two] old six type renamed to rtloader

[rtloader][three] old six type renamed to rtloader

[rtloader][cgo] fix pympler helper from depcreated six naming

* Gopkg.lock was out of sync

* fix nightly build - integrations from master

julien-lebot committed:
Ensure that check is not JMX before running it (#4031)

* Ensure that check is not JMX before running it

* Added missing import

* Simplify logic

* Added release note

* Addressed comments

* Changed "deprecated" -> "not supported" in release note

* Changed wording of release note.

truthbk committed:
Revert "[omnibus][python2] shared build (#3554)" (#4057)

This reverts commit 8a83c799136b0d3ca925b5770cbdc1330f42ec8a.

truthbk committed:
[status] fixing unmarshaling reference bug (#4058)

aerostitch committed:
Doc: add a mention to the DD_CLUSTER_AGENT_URL variable (#4051)

albertvaka committed:
Upgrade jmxfetch (#4056)

blemale committed:
[logs] Release file descriptor on file rotation when tailer is stuck (#4055)

* [logs] Rename tl into tailer in tailer_test.go

* [logs] Close tailer on file rotation even when stuck

Currently if the sender could not send the logs because it can't establish a
connection to the logs intake, the whole pipeline is stuck. Thus when there is
a file rotation the previous tailer is never close and it keeps a file
descriptor causing a leak.

This commit makes possible to close a tailer even if it is stuck thanks to the
use of a context.

Commit:
fix host metadata on GKE (#3907)

cswatt committed:
add note about host tags format (#3639)

* add note about host tags format

* adding to config file, surfacing link higher

* remove line

* add environment variable line

* Update docs/agent/changes.md

Co-Authored-By: ruthnaebeck <19349244+ruthnaebeck@users.noreply.github.com>

truthbk committed:
[omnibus] python2 shared-libs (again) w/ new builders (#4061)

* Revert "Revert "[omnibus][python2] shared build (#3554)" (#4057)"

This reverts commit c377f34a212187082b6a7ad28ab9ae4e28f9736a.

* [gitlab] use updated builder

hush-hush committed:
Fix python bin path for linting (#4052)

albertvaka committed:
Add 'Origin: datadog' to staging apt repo (#4047)

This field is used to whitelist repos (eg: for unattended upgrades), and was requested by a customer.

The deb-s3 tool we use has support for adding this field: https://github.com/krobertson/deb-s3/blob/master/README.md

Example of the Origin field used on a repo: http://repository-origin.spotify.com/dists/stable/Release

derekwbrown committed:
[windows] when uninstalling in an environment where a domain user was (#4053)

used as the dd agent user context, do not attempt to delete the user

AlexandreYang committed:
Improve systemd configuration and metric names (#3974)

upgrade:
- Update to the configuration of the systemd check: ``unit_names`` is now required and only matching units will be monitored, ``unit_regexes`` configuration has been removed.
- Several metrics sent by the systemd check have been renamed. The integration is now stable.

enhancements:
- Add `private_socket` configuration to the systemd check. Defaults to `/run/systemd/private` (or `/host/run/systemd/private` when using Docker Agent).

achntrl committed:
Get more information from windows events (#3987)

* Get more information from windows events

* Display a truncated flag is the message is truncated

* Nits: curly bracket for if blocks and HN

KSerrania committed:
[omnibus][python3] Remove embedded Python 3 test directory (#4063)

The python3.7/test contains source files and compiled files for
core Python libraries' tests. We remove them for Python 2, and we should
do the same for Python 3 (installed size decreases by 60MB for the MacOS dmgs, 20MB for the deb and rpm packages).

ahmed-mez committed:
Extract and expose CLC runner stats (#3931)

KSerrania committed:
[kitchen] Add tests to check that the Agent does start with Python 3 (#4033)

Sanity checks to verify that we do include python3 and that the
Agent can restart in python2 or python3 mode.

KSerrania committed:
Revert "Remove yum warnings while removing the Agent package. (#3781)" (#4065)

Revert the yum maintainer scripts change to remove warnings when doing yum remove datadog-agent, as the new logic also removes manually installed packages, which shouldn't happen.

mfpierre committed:
Enhance short lived container autodiscovery (#4048)

* Enhance short lived container autodiscovery + rework docker config provider

* Fix comments

mfpierre committed:
Poll AD docker config more frequently (#3930)

mfpierre committed:
Add the histogram bucket bridge (#3937)

* Add the histogram bucket bridge

* Introduce MetricSampleContext interface

* Store bucket deltas

* Handle arbitrarily infinity buckets

* Add benchmark

* Optimize using InsertN into the sketch

* Change infinity logic + add tests

* Consider monotonic bucket parameter

KSerrania committed:
Add python_version to template config file (#4013)

Adds the python_version configuration option to the template configuration file.

albertvaka committed:
Disable linter when the agent runs in python3 (#4067)

xlucas committed:
Use OpenJDK11 in the docker image (#4034)

truthbk committed:
[omnibus] strip symbols (#3759)

* [omnibus] strip symbols - linux - wip

* [omnibus] splitting strip steo - previous attempt was a NOP

* [omnibus][ostool] actual debug statement for symbol stripping

* [gitlab] omnibus build on debug log-level for troubleshooting purposes

* [gitlab][skip ci] no need for debug levels

* [strip] comment some stuff out

* [omnibus] omnibus-ruby now responsible for stripping logic

* [strip+debug] stripping logic in omnibus-ruby + dbg install scripts

* [strip+dbg] empty dbg install scripts + trigger stripping and debug build

* [omnibus] some cleanup, heavy lifting in omnibus-ruby

* [omnibus] some cleanup, heavy lifting in omnibus-ruby

* [omnibus] some more cleanup

* [omnibus] cleaner debug build at the project level

* [release] restore omnibus-ruby branch

* [gitlab] addressing multi-artifact scenarios

[gitlab] fixing pipelines

* [dockerfile] build should ignore dbg deb artifact

* [omnibus] don't strip psycopg2

* [omnibus] clarifying comment

* [dockerfile] use egrep to succeed if we got matches.

permission denied errors can break the chain, egrep will
return 0 if we got any matches in the find, which will
allow the chain of commands to continue,

[dockerfile] extract the verbose way for proper processing

leeavital committed:
Log issues with networkID at Info level (#4062)

The networkID flag is optional, and it's okay if not detected.

derekwbrown committed:
Fix configuration parameter for process & trace so that spaces in the (#4064)

filename are OK

derekwbrown committed:
[windows] adds missing system.mem.pagefile.* stats back (#4040)

* [windows] adds missing system.mem.pagefile.* metrics back (were present
in A5, missing in A6)

Update unit tests accordingly

* Add information in changes.md

* Update docs/agent/changes.md

Co-Authored-By: Albert Vaca <albert.vaca@datadoghq.com>

DylanLovesCoffee committed:
add connection blacklisting (#4039)

* add connection blacklisting

* fix issues and attempt to optimize

* add release note and deps. Also minor format edits

* change ports to a map and create blacklist store

* fix comment

* fix bad test

* refactoring

* support edge cases

* change config name and add edge cases to tests

* refactor tracer statements

* add config unit test and have bad filters log errors

* fix bad formatting

albertvaka committed:
[release] Update release.json with 6.14.0-rc.1 entry (#4070)

albertvaka committed:
Revert "Add 'Origin: datadog' to staging apt repo (#4047)" (#4072)

This reverts commit d3c9123ac307411dbb3c63775b5251eeecf8bdb2.

p-lambert committed:
[ebpf] Fix offset guessing process when IPv6 is disabled (#4075)

KSerrania committed:
[pkg/collector/python] Log python linter warnings at debug level (#4071)

Log Python3 linter warnings as debug level instead of warn level.
Warn is not the correct level for this, as there could be hundreds of linter warnings for each check. 
Also, add a missing release note.

albertvaka committed:
[release] Update release.json with 6.14.0-rc.2 entry (#4079)

xlucas committed:
Update release workflow (#4074)

* New temporary workflow for agent 6 using only one tag

* Update e2e tests job

truthbk committed:
[agent] check command: fix JMX validation (#4081)

* [agent] check command: fix JMX validation

* [agent] check command: fix JMX validation - bail out sooner

p-lambert committed:
[process] Add RTT fields to agent payload (#4084)

ajacquemot committed:
Fixed a race condition for docker logs for short lived containers (#4085)

* Fixed a race condition for docker logs for short lived containers

* Improved comment for reader close

truthbk committed:
[memtrack] performance optimization - dont get stack unless necessary (#4087)

* [memtrack] performance optimization - dont get stack unless necessary

* [memtrack] performance optimization - dont get stack unless necessary

* [memtrack] use native seelog types for loglevel reporting

* [log] getting the log level must be thread-safe

albertvaka committed:
[release] Update release.json with 6.14.0-rc.3 entry (#4088)

joyzhang-dd committed:
[Logs]: Increase the HTTP batch size to provide larger logging throughput (#4090)

albertvaka committed:
Remove default log-level arg from jmx command (#4095)

So the env var is used if the arg is not specified. Changed the default
to 'debug' to not change the behaviour of already installed agents.

truthbk committed:
[memtrack] two and three need to have their own references to callback (#4093)

* [rtloader] pass memory tracking facilities down to two and three

* [rtloader] clang formatting fixes

* [rtloader] memtrack: adding a little bit of safety for get/set of callback

* [rtloader] memtrack: fixing windows typo

* [rtloader] memory barrier __sync_synchronize() should be defined on win (gcc)

shang-wang committed:
Add platform detection using uname command (#4091)

* add detect platform using uname

* fix

* add to missing error text

kevinconaway committed:
Fix upstart scripts to not stop the system-probe when agent stops (#4083)

* Fix upstart scripts to start system-probe when agent starts

* Don't start the sysprobe when the agent starts.  Instead, don't _stop_ it when the agent stops

KSerrania committed:
[release] Update release.json with 6.14.0-rc.4 entry (#4098)

achntrl committed:
Remove spammy log when we fail formatting windows events (#4097)

xlucas committed:
Rework tagging plan for agent 6 & 7 (#4099)

derekwbrown committed:
Fix feature that doesn't remove domain user; fixes case where user (#4094)

provided a password but not a username (so we used the default username
on the domain)

albertvaka committed:
[release] Update release.json with 6.14.0-rc.5 entry (#4102)

hush-hush committed:
Log error when the base class can't be found or strings can't be initialized.

This is to match the behavior or three in two

albertvaka committed:
Improve wording of error message (#4101)

The previous error message didn't seem accurate enough to me.

xlucas committed:
Revert "Use OpenJDK11 in the docker image (#4034)" (#4105)

This reverts commit 611b939a522bd307ecc42ee6109aff2f0742902e.

KSerrania committed:
[release] Update release.json with 6.14.0-rc.6 entry (#4109)

truthbk committed:
[loader] python: synchronous-serial linting/validation (#4113)

* [loader] python: synchronous-serial linting validation

* [config] setting more appropriate values + documenting option

* Apply suggestions from code review

Co-Authored-By: Ofek Lev <ofekmeister@gmail.com>

* Update pkg/config/config_template.yaml

Co-Authored-By: Albert Vaca <albert.vaca@datadoghq.com>

therve committed:
Bump a7 version (#4114)

This will speed up validation significantly.

mfpierre committed:
APM: initialize root.Meta tag when container tags are applied to avoid nil pointer panic (#4119)

* Fix nil map

shang-wang committed:
Log instead of error out when process-agent is not enabled in config in MacOS (#4115)

Currently the process corecheck returns an error when process-agent is disabled in the config. This results in process-agent showing up in error status in datadog-agent UI. The PR logs the message instead and return no error, so it shouldn't confuse customers anymore.

KSerrania committed:
[release] Update release.json with 6.14.0-rc.7 entry (#4120)

shang-wang committed:
[process-agent] improve comment and log message (#4122)

* Only log instead of error out when process-agent is not enabled in
config

* remove nil

* improve comment and log message

* nit

ruthnaebeck committed:
[Doc] Integration Management (#4104)

* [Doc] Integration Management

* wording update

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

KSerrania committed:
Add release note for Python 3 in Agent (#4103)

Add release note for Python 3 in Agent.
Python 3 will be included in the Agent in 6.14.0. This needs to be documented in the changelog.

Co-Authored-By: Dhruv Sahni <dhruv.sahni@datadoghq.com>

KSerrania committed:
Add release note for RPM key rotation (#4111)

Add release note for the RPM key rotation in Agent.
The RPM key used to sign our RPM packages has been rotated. This is a potentially breaking change (can prevent install of Agent packages) which needs to be added to the changelog.

ruthnaebeck committed:
[Doc] Hostname resolution (#4117)

* [Doc] Hostname resolution

* add comment

* re-run tests

* re-run tests

Commit:
Migrating configuration file to the new norm (#3917)

* migrating configuration file to the new norm

* GLITCHES fix

Co-Authored-By: ruthnaebeck <19349244+ruthnaebeck@users.noreply.github.com>

* removing legacy parameters

* wording updates

Co-Authored-By: FlorianVeaux <florian.veaux@datadoghq.com>

* review update

* removing tagging section

* spelling fix

* adding release note

* Removing python options

* Removing python option

* Removing python options

* removing unsupported parameter

clamoriniere committed:
[Fix][AutoConfig] update check in case of an empty tagger during the … (#4127)

* [Fix][AutoConfig] update check in case of an empty tagger during the initial configuration
* add releasenote

Commit:
Removing upgrade documentation (#4112)

* Removing upgrade documentation

Removing upgrade documentation from the repo, and leaving a placeholder to point to the public documentation page instead.

* wording update

* wording update

Co-Authored-By: Kaylyn <kaylyn.sigler@datadoghq.com>

* removing trailing lines

clamoriniere committed:
[Fix] tagger cache miss overwrite (#4133)

* fix tagger update with TagInfo empty from cache-miss

KSerrania committed:
[omnibus] Remove Python 3 integrations on upgrade/uninstall (#4137)

truthbk committed:
[omnibus] cffi_backend stripping break SO on CentOS7 (#4135)

albertvaka committed:
[release] Update release.json with 6.14.0-rc.8 (#4138)

leeavital committed:
Roll back to building system-probe in its own image (#4147)

This works around conntrack issues that arise when building with go
1.12, and should be rolled back once system-probe can be built with 1.12

mfpierre committed:
Add tag hash for all services (#4146)

ruthnaebeck committed:
[Doc] Agent status page (#4141)

truthbk committed:
[release] Update release.json with 6.14.0-rc.9

Commit:
Removing duplicated documentation (#4132)

julien-lebot committed:
Make NTPCheck use CheckBase's MinCollectionInterval (#4145)

* Change NTP handling of collection interval to use the BaseCheck collection interval instead.

julien-lebot committed:
Revert "Make NTPCheck use CheckBase's MinCollectionInterval (#4145)" (#4150)

This reverts commit 7baadfd3389d07c8d5cd93f06fcaacb266c5ce70.

KSerrania committed:
[kitchen] Do not allow job failure (#4151)

derekwbrown committed:
Add region tag to ssm command (#4152)

KSerrania committed:
Add entry and prelude for 6.14.0 release (#4124)

ruthnaebeck committed:
[Doc] GUI (#4140)

KSerrania committed:
Update CHANGELOG for 6.14.0 (#4153)

mfpierre committed:
Fix diagnose config setup (#4155)

gbbr committed:
pkg/trace/traceutil: add SetMeta function (#4131)

Follow up to #4119

gbbr committed:
pkg/trace/obfuscate: ensure double-quoted strings after assignments are obfuscated (#4100)

This change ensures that double-quoted strings after assignments get
obfuscated. In general, double-quoted strings can denote identifiers in
some cases, e.g.:

> If the ANSI_QUOTES SQL mode is enabled, string literals can be quoted only within single quotation marks because a string quoted within double quotation marks is interpreted as an identifier.
Ref: https://dev.mysql.com/doc/refman/8.0/en/string-literals.html

Using double quotes for strings is generally considered bad practice but
in some cases is omitted. It should be safe, until proven wrong, to
consider double quoted strings following an assignments to be a token
that needs to be obfuscated.

ogaca-dd committed:
Fix Win32 build. (#4096)

albertvaka committed:
Make agent_log take a const string (#4073)

So we can use it to log string literals.

albertvaka committed:
Simplify get_multi_python_location (#4080)

Expression `rtloader_root or embedded_path` was within an `if rtloader_root is None`

albertvaka committed:
Propagate errors from python to the logs (#4110)

albertvaka committed:
Remove unused param (#4157)

ogaca-dd committed:
Fix rtloader.clear_cmake_cache (#4121)

julien-lebot committed:
RE: Make NTPCheck use CheckBase's MinCollectionInterval (#4162)

* Change NTP handling of collection interval to use the BaseCheck collection interval instead.

julien-lebot committed:
Fix NTP check tagging (#4144)

* Move CommonConfigure after BuildID so that CheckBase's Tags are set on the correct sender.

ofek committed:
Stop shipping tokumx on Python 3 (#4164)

sunhay committed:
Add logging on inactivity for system-probe to help customers debug (#4169)

derekwbrown committed:
Merge branch 'db/localservice' of github.com:DataDog/datadog-agent into db/localservice

derekwbrown commented:
Superceded by https://github.com/DataDog/datadog-agent/pull/4179
Had diverged too far from master for clean PR
